File: src/main/java/com/github/J1372/WebBackend/Dto/Color.java
package com.github.J1372.WebBackend.Dto;

public enum Color {
    WHITE,
    BLACK,
    EITHER;

    public static Color fromChar(char character) {
        return switch (character) {
            case 'w' -> Color.WHITE;
            case 'b' -> Color.BLACK;
            default -> Color.EITHER;
        };
    }
}

File: src/main/gameplay/src/color.ts
export enum Color {
    WHITE,
    BLACK,
}

export namespace Color {
    export function opposite(color: Color) {
        return color === Color.WHITE ? Color.BLACK : Color.WHITE;
    }

    export function toString(color: Color) {
        if (color === Color.WHITE) {
            return 'White';
        } else if (color === Color.BLACK) {
            return 'Black';
        } else {
            return '';
        }
    }
}